File: myfs/lib/models/schedule.dart
/// Lịch học / thời khóa biểu (LichHoc) – khớp entity Schedule bên backend.
class Schedule {
  final int id;
  final int classId;
  final int dayOrder; // 2=Thứ 2 ... 7=Thứ 7, 8=CN
  final int period;
  final String subject;
  final String? room;
  final String? teacherName;
  final String? startTime;
  final String? endTime;

  const Schedule({
    required this.id,
    required this.classId,
    required this.dayOrder,
    required this.period,
    required this.subject,
    this.room,
    this.teacherName,
    this.startTime,
    this.endTime,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) => Schedule(
    id: json['id'] as int,
    classId: json['classId'] as int,
    dayOrder: json['dayOrder'] as int,
    period: json['period'] as int,
    subject: json['subject'] as String,
    room: json['room'] as String?,
    teacherName: json['teacherName'] as String?,
    startTime: json['startTime'] as String?,
    endTime: json['endTime'] as String?,
  );

  /// Nhãn thứ trong tuần.
  static String dayLabel(int dayOrder) {
    switch (dayOrder) {
      case 2:
        return 'Thứ 2';
      case 3:
        return 'Thứ 3';
      case 4:
        return 'Thứ 4';
      case 5:
        return 'Thứ 5';
      case 6:
        return 'Thứ 6';
      case 7:
        return 'Thứ 7';
      case 8:
        return 'Chủ nhật';
      default:
        return 'Thứ $dayOrder';
    }
  }
}

File: backend/src/main/java/com/myfs/backend/model/Schedule.java
package com.myfs.backend.model;

import jakarta.persistence.*;

/**
 * Entity ánh xạ bảng schedule – lịch học / thời khóa biểu (LichHoc).
 * dayOrder: 2=Thứ 2 ... 7=Thứ 7, 8=Chủ nhật.
 */
@Entity
@Table(name = "schedule")
public class Schedule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "class_id", nullable = false)
    private Integer classId;

    @Column(name = "day_order", nullable = false)
    private Integer dayOrder;

    @Column(name = "period", nullable = false)
    private Integer period;

    @Column(name = "subject", nullable = false, length = 100)
    private String subject;

    @Column(name = "room", length = 50)
    private String room;

    @Column(name = "teacher_name", length = 100)
    private String teacherName;

    @Column(name = "start_time", length = 10)
    private String startTime;

    @Column(name = "end_time", length = 10)
    private String endTime;

    public Schedule() {}

    public Integer getId()          { return id; }
    public Integer getClassId()     { return classId; }
    public Integer getDayOrder()    { return dayOrder; }
    public Integer getPeriod()      { return period; }
    public String  getSubject()     { return subject; }
    public String  getRoom()        { return room; }
    public String  getTeacherName() { return teacherName; }
    public String  getStartTime()   { return startTime; }
    public String  getEndTime()     { return endTime; }

    public void setId(Integer v)          { this.id = v; }
    public void setClassId(Integer v)     { this.classId = v; }
    public void setDayOrder(Integer v)    { this.dayOrder = v; }
    public void setPeriod(Integer v)      { this.period = v; }
    public void setSubject(String v)      { this.subject = v; }
    public void setRoom(String v)         { this.room = v; }
    public void setTeacherName(String v)  { this.teacherName = v; }
    public void setStartTime(String v)    { this.startTime = v; }
    public void setEndTime(String v)      { this.endTime = v; }
}
